### PR TITLE
updated test data and added integration tests to cover known use case…

### PR DESCRIPTION
**Pull request by RobertLaverick**
…s for EnsureGuaranteeService

### CODE REVIEW

**Review by crystalmark — APPROVED** [review]

File: it/services/EnsureGuaranteeServiceIntegrationSpec.scala [hunk 1]
```diff
@@ -7,77 +7,184 @@ import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
...
-      val validator = app.injector.instanceOf[XmlValidationService]
-
-      val result = service.ensureGuarantee(TestData.buildXml(TestData.standardInputXML))
-
-      validator.validate(result.right.get.toString().filter(_ > ' '), DepartureDeclarationXSD) mustBe a[Right[_, XmlValid]]
+  "listed scenarios" - {
+    "GB -> EU, type 1, CAL w/o Reference Number - SM Passes through to core unedited, new one added" in {
+      val insertXml = TestData.basicGuarantee ++ TestData.goodsWithCustomSpecialMention(<SPEMENMT2><AddInfCodMT23>CAL</AddInfCodMT23></SPEMENMT2>)
+      val xml = TestData.buildGBEUXml(insertXml)
+      val expectedXml = TestData.buildGBEUXml(TestData.basicGuarantee ++ TestData.goodsWithCustomSpecialMention(<SPEMENMT2><AddInfCodMT23>CAL</AddInfCodMT23></SPEMENMT2><SPEMENMT2><AddInfMT21>1.00GBP21GB0000010000HU1</AddInfMT21><AddInfCodMT23>CAL</AddInfCodMT23></SPEMENMT2>))
```
Comment (DavidGregory084) [inline it/services/EnsureGuaranteeServiceIntegrationSpec.scala:75]: Could we possibly extract some of these large XML nodes into their own `val`s and format them so they aren't on one line?

**Review by DavidGregory084 — APPROVED** [review]
Looks good! 👍

File: it/services/EnsureGuaranteeServiceIntegrationSpec.scala (hunk 1 above)
Comment (RobertLaverick) [reply it/services/EnsureGuaranteeServiceIntegrationSpec.scala:75]: Will do

**Review by DavidGregory084 — APPROVED** [review]
💯

**Pull request merged**